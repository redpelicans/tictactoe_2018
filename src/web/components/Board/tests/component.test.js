import React from 'react';
import { shallow } from 'enzyme';
import { BoardPanel, Board, Cell, Message } from '../';

const state = {
  status: 'GAME_OVER',
  board: ['x', 'x', 'x'],
  currentPlayer: { name: 'Toto' },
  computerPlay: new Function(),
};

describe('<Board/>', () => {
  it('should render', () => {
    const wrapper = shallow(<BoardPanel {...state} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a <Cell/>', () => {
    const wrapper = shallow(<Board {...state} />).find(Cell);
    expect(wrapper).toHaveLength(3);
  });

  it('should render a <Message/>', () => {
    const wrapper = shallow(<Message {...state} />);
    expect(wrapper.find('span').text()).toMatch(new RegExp(state.currentPlayer.name));
  });

  it('should render a <BoardPanel/>', () => {
    expect(shallow(<BoardPanel {...state} />).find(Message)).toHaveLength(1);
    expect(shallow(<BoardPanel {...state} />).find(Board)).toHaveLength(1);
  });

  it('should render a <Cell/>', () => {
    const text = shallow(<Cell piece="X" />)
      .childAt(0)
      .text();
    expect(text).toEqual('X');
  });

  it('should render an empty <Cell/>', () => {
    const text = shallow(<Cell />)
      .childAt(0)
      .text();
    expect(text).toEqual('\u00a0');
  });
});
