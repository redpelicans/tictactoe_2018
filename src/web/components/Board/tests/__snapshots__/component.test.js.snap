// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Board/> should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    board={
      Array [
        "x",
        "x",
        "x",
      ]
    }
    computerPlay={[Function]}
    currentPlayer={
      Object {
        "name": "Toto",
      }
    }
    status="GAME_OVER"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "bsClass": "container",
      "children": Array [
        <Row
          bsClass="row"
          componentClass="div"
        >
          <Unknown
            currentPlayer={
              Object {
                "name": "Toto",
              }
            }
          />
        </Row>,
        <Row
          bsClass="row"
          componentClass="div"
        >
          <Unknown
            board={
              Array [
                "x",
                "x",
                "x",
              ]
            }
            computerPlay={[Function]}
            status="GAME_OVER"
          />
        </Row>,
      ],
      "className": "panel",
      "componentClass": "div",
      "fluid": false,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "row",
          "children": <Unknown
            currentPlayer={
              Object {
                "name": "Toto",
              }
            }
          />,
          "componentClass": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "currentPlayer": Object {
              "name": "Toto",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "row",
          "children": <Unknown
            board={
              Array [
                "x",
                "x",
                "x",
              ]
            }
            computerPlay={[Function]}
            status="GAME_OVER"
          />,
          "componentClass": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "board": Array [
              "x",
              "x",
              "x",
            ],
            "computerPlay": [Function],
            "status": "GAME_OVER",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsClass": "container",
        "children": Array [
          <Row
            bsClass="row"
            componentClass="div"
          >
            <Unknown
              currentPlayer={
                Object {
                  "name": "Toto",
                }
              }
            />
          </Row>,
          <Row
            bsClass="row"
            componentClass="div"
          >
            <Unknown
              board={
                Array [
                  "x",
                  "x",
                  "x",
                ]
              }
              computerPlay={[Function]}
              status="GAME_OVER"
            />
          </Row>,
        ],
        "className": "panel",
        "componentClass": "div",
        "fluid": false,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "row",
            "children": <Unknown
              currentPlayer={
                Object {
                  "name": "Toto",
                }
              }
            />,
            "componentClass": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "currentPlayer": Object {
                "name": "Toto",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "row",
            "children": <Unknown
              board={
                Array [
                  "x",
                  "x",
                  "x",
                ]
              }
              computerPlay={[Function]}
              status="GAME_OVER"
            />,
            "componentClass": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "board": Array [
                "x",
                "x",
                "x",
              ],
              "computerPlay": [Function],
              "status": "GAME_OVER",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
